module/icp/asm-aarch64/blake3/blake3_neon.S

```diff
@@ -0,0 +1,637 @@
+	eor	v19.16b, v2.16b, v19.16b
+	add	v6.4s, v6.4s, v0.4s
+	ushr	v24.4s, v19.4s, #12
+	shl	v19.4s, v19.4s, #20
+	eor	v4.16b, v6.16b, v4.16b
+	orr	v19.16b, v19.16b, v24.16b
+	ushr	v24.4s, v4.4s, #16
+	shl	v4.4s, v4.4s, #16
+	orr	v4.16b, v4.16b, v24.16b
+	add	v1.4s, v4.4s, v1.4s
+	ldrb	w9, [x8, #9]
+	eor	v0.16b, v1.16b, v0.16b
+	ushr	v24.4s, v0.4s, #12
+	shl	v0.4s, v0.4s, #20
+	orr	v0.16b, v0.16b, v24.16b
+	stp	q22, q16, [x0]
+	stp	q23, q7, [x0, #160]
+	stp	q5, q3, [x0, #192]
+	stp	q18, q6, [x0, #32]
+	stp	q17, q19, [x0, #96]
+	stp	q4, q21, [x0, #224]
+	stp	q2, q1, [x0, #128]
+	stp	q0, q20, [x0, #64]
+	ldr	q24, [x1, x9, lsl #4]
+	ldrb	w9, [x8, #11]
+	add	v22.4s, v24.4s, v22.4s
+	str	q22, [x0]
+	ldr	q24, [x1, x9, lsl #4]
+	ldrb	w9, [x8, #13]
+	add	v22.4s, v22.4s, v20.4s
+	eor	v21.16b, v22.16b, v21.16b
+	add	v16.4s, v24.4s, v16.4s
+	ushr	v25.4s, v21.4s, #8
+	shl	v21.4s, v21.4s, #24
+	str	q16, [x0, #16]
+	orr	v21.16b, v21.16b, v25.16b
+	ldr	q25, [x1, x9, lsl #4]
+	ldrb	w8, [x8, #15]
+	add	v23.4s, v21.4s, v23.4s
+	eor	v20.16b, v23.16b, v20.16b
+	add	v18.4s, v25.4s, v18.4s
+	str	q18, [x0, #32]
+	add	v16.4s, v16.4s, v17.4s
+	add	v18.4s, v18.4s, v19.4s
+	ldr	q25, [x1, x8, lsl #4]
+	ushr	v24.4s, v20.4s, #7
+	shl	v20.4s, v20.4s, #25
+	eor	v5.16b, v16.16b, v5.16b
+	eor	v3.16b, v18.16b, v3.16b
+	orr	v20.16b, v20.16b, v24.16b
+	ushr	v24.4s, v5.4s, #8
+	shl	v5.4s, v5.4s, #24
+	stp	q22, q16, [x0]
+	ushr	v16.4s, v3.4s, #8
+	shl	v3.4s, v3.4s, #24
+	orr	v5.16b, v5.16b, v24.16b
+	orr	v3.16b, v3.16b, v16.16b
+	stp	q5, q3, [x0, #192]
+	add	v2.4s, v3.4s, v2.4s
+	add	v3.4s, v25.4s, v6.4s
+	add	v7.4s, v5.4s, v7.4s
+	eor	v5.16b, v2.16b, v19.16b
+	add	v3.4s, v3.4s, v0.4s
+	ushr	v6.4s, v5.4s, #7
+	shl	v5.4s, v5.4s, #25
+	stp	q18, q3, [x0, #32]
+	eor	v3.16b, v3.16b, v4.16b
+	orr	v4.16b, v5.16b, v6.16b
+	ushr	v5.4s, v3.4s, #8
+	shl	v3.4s, v3.4s, #24
+	orr	v3.16b, v3.16b, v5.16b
+	add	v1.4s, v3.4s, v1.4s
+	eor	v17.16b, v7.16b, v17.16b
+	eor	v0.16b, v1.16b, v0.16b
+	ushr	v24.4s, v17.4s, #7
+	shl	v17.4s, v17.4s, #25
+	stp	q2, q1, [x0, #128]
+	ushr	v1.4s, v0.4s, #7
+	shl	v0.4s, v0.4s, #25
+	stp	q23, q7, [x0, #160]
+	orr	v7.16b, v17.16b, v24.16b
+	orr	v0.16b, v0.16b, v1.16b
+	stp	q7, q4, [x0, #96]
+	stp	q3, q21, [x0, #224]
+	stp	q0, q20, [x0, #64]
+	ret
+.Lfunc_end4:
+	.size	round_fn4, .Lfunc_end4-round_fn4
+                                        // -- End function
+
+	.type	MSG_SCHEDULE,@object            // @MSG_SCHEDULE
+MSG_SCHEDULE:
+	.ascii	"\000\001\002\003\004\005\006\007\b\t\n\013\f\r\016\017"
+	.ascii	"\002\006\003\n\007\000\004\r\001\013\f\005\t\016\017\b"
+	.ascii	"\003\004\n\f\r\002\007\016\006\005\t\000\013\017\b\001"
+	.ascii	"\n\007\f\t\016\003\r\017\004\000\013\002\005\b\001\006"
+	.ascii	"\f\r\t\013\017\n\016\b\007\002\005\003\000\001\006\004"
+	.ascii	"\t\016\013\005\b\f\017\001\r\003\000\n\002\006\004\007"
+	.ascii	"\013\017\005\000\001\t\b\006\016\n\002\f\003\004\007\r"
+	.size	MSG_SCHEDULE, 112
+
+	.ident	"Debian clang version 11.0.1-2"
```

Comment:
I'd say this can probably go into the comment at the top, the branding's irrelevant here

Reply:
@nabijaczleweli - Just for info, this is "Work in progress" ...
I will split this thing up into seperate commits... it's not generelly finished :)